File: src/ProjectR.Scripting/content/scripts/afflictions/Buffs/Clearcasting.lua
function GetName()
  return "Clearcasting"
end

function GetType()
  return Buff
end

function GetHookPoints()
  return { UsingMP }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint == UsingMP then
    param1 = 0
  end
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Buffs/Clearcasting.cs
using ProjectR.Interfaces.Model;

namespace ProjectR.Scripting.Afflictions
{
    public class Clearcasting : Affliction
    {
        public override string Name { get { return "Clearcasting"; } }

        public override AfflictionType Type { get { return AfflictionType.Buff; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new HookPoint[]
                {
                    HookPoint.UsingMP, 
                };
            }
        }

        protected override void OnUsingMP(ICharacter character, ref double value)
        {
            value = 0;
        }
    }
}